tools/mem_write.py

```diff
@@ -9,17 +10,28 @@ def main():
     # List all debuggable TREZORs
     devices = [device for device in enumerate_devices() if hasattr(device, 'find_debug')]
 
+    # Try to get debug transport on devices
+    debug_transport = None
+    for dev in devices:
+        try:
+            transport = dev
+            debug_transport = dev.find_debug()
+            break
+        except Exception:
+            pass
+
     # Check whether we found any
     if len(devices) == 0:
         print('No TREZOR found')
         return
 
-    # Use first connected device
-    transport = devices[0]
-    debug_transport = devices[0].find_debug()
+    # Check if its available
+    if not debug_transport:
```

Comment:
is there any case where you don't get an exception from the debug device, but get a falsey value for `find_debug`?

Reply:
> is there any case where you don't get an exception from the debug device, but get a falsey value for `find_debug`?

I always either get a valid handle or `TransportException`, but I never get `False`
```
bridge:lib02
webusb:001:1:2
```
Any call to `device['webusb'].find_debug()` raises:
```
TransportException: Debug device not available
```

As long as the enumerator grabs bridge device fist, everything works, but if it happens upon the webusb first, it fails.  This is what happened when I originally made the PR.  I recently noticed that `trezord.exe` had abended, so perhaps this is what I was seeing.

I could look at linux behaviour later, perhaps it is more resilient.

Reply:
oooh I understand what you meant now. No, this is not a good way to do that. These scripts are really old.

Instead do something like this:

```python
def get_debug_device():
	for device in enumerate_devices():
		try:
			return TrezorClientDebugLink(device)
		except Exception:
			pass
	raise Exception("no debuggable device found")

def main():
	# ....
	client = get_debug_device()
	client.debug.memory_write(...)
```

Reply:
Will do... I'll update it in the next day or so.

Thx

Reply:
Does `DebugLink` require debug firmware loaded on the device?  I can't seem to get any `DebugLink` method to run.

I [collected logs](https://gist.github.com/brianddk/a735927014e4024b10edc4efacd01026) for running `py.test trezorlib/tests/device_tests/test_msg_getaddress.py` compared to `py.test trezorlib/tests/device_tests/test_msg_getaddress.py`.  The `py.test` run was failing in DebugLink.

Reply:
yes, debuglink requires a debug firmware.
pytest should be able to run on a debugless firmware if you `export INTERACT=1`